docs/source/daml/reference/contract-keys.rst

```diff
@@ -19,76 +19,76 @@ Here's an example of setting up a contract key for a bank account, to act as a b
 What can be a contract key
 **************************
 
-The key can be an arbitrary expression that does **not** contain contract IDs. However, it **must** include every party that you want to use as a ``maintainer`` (see `Specifying maintainers`_ below).
+The key can be an arbitrary serializable expression that does **not** contain contract IDs. However, it **must** include every party that you want to use as a ``maintainer`` (see `Specifying maintainers`_ below).
 
 It's best to use simple types for your keys like ``Text`` or ``Int``, rather than a list or more complex type.
 
 Specifying maintainers
 **********************
 
-If you specify a contract key for a template, you must also specify a ``maintainer`` or maintainers, in a similar way to specifying signatories or observers.  However, maintainers are computed from the ``key`` instead of the template arguments.  In the example above, the ``bank`` is ultimately the maintainer of the key.  Maintainers are the parties that know about all of the keys that they are party to, and are used by the engine to guarantee uniqueness of contract keys.  The maintainers **must** be signatories of the contract.
+If you specify a contract key for a template, you must also specify a ``maintainer`` or maintainers, in a similar way to specifying signatories or observers. The maintainers "own" the key in the same way the signatories "own" a contract. Just like signatories of contracts prevent double spends or use of false contract data, maintainers of keys prevent double allocation or incorrect lookups. Since the key is part of the contract, the maintainers **must** be signatories of the contract. However, maintainers are computed from the ``key`` instead of the template arguments.  In the example above, the ``bank`` is ultimately the maintainer of the key. 
```

Comment:
To me, that's an issue ledger fidelity. In kvutil ledgers, the participants have simply delegated their responsibility / right to validate transactions to centralized validating nodes. I still think the maintainers conceptually own the index / keys and have the responsibility to maintain uniqueness.

Think of a deployment situation where parties can be hosted on multiple Participants on different domains. It is exactly the maintainers responsibility not to allocate keys on different domains and thus violate uniqueness.

Reply:
Another way of thinking about it is this:

In an idealised DAML Ledger where Parties correspond to Participants and DAML's privacy model is fully implemented, if all signatories collaborate, they can double spend their contracts, for nobody else knows about these contracts at all.

If all maintainers collaborate, they can double-allocate a key, for if C1 has key K with signatories M u S and maintainers M, they may be able to create a disjoint set S' and create C2 with key K, signatories M u S' and maintainers M. Since only M know about C1 and C2, nobody is any the wiser.

Reply:
Sure, but this presupposes that signatories/maintainers collaborate in some way during validation. I can also take your example and consider the case where my participant node just chooses to show me some contracts as double-spent locally, without any of the other signatories' participants being any the wiser. Can now any node double spend the contract? Should we talk about that in the DAML docs too? 

What I'm trying to say, there are many ways to slice it, the question is which ones are the most useful/understandable. While validation is part of the process of every implementation, I think it's confusing to talk about it at the level of the language/abstract ledger model, where the details of the validation process are unclear.